targets/TARGET_STM/hal_tick_overrides.c

```diff
@@ -27,11 +27,10 @@ extern int mbed_sdk_inited;
 void init_16bit_timer(void);
 void init_32bit_timer(void);
 
-#if TIM_MST_BIT_WIDTH == 16
 // Variables also reset in us_ticker_init()
+uint32_t total_ticks = 0;
```

Comment:
If retaining these variables as global, please give them some sort of prefix to minimise chance of namespace collisions.

But you can make them static. As I said, I don't think they need manual init, but even then, this current code makes no sense. Why is `init_16bit_timer` setting them to zero, rather than `HAL_InitTick` itself? Do it there, and they can be static.